tests/test_component_media.py

```diff
@@ -282,6 +287,437 @@ class Media:
+            <script src="path/to/script2.js"></script>
+            """,
+        )
+
+    def test_bytes(self):
+        """
+        Test that media work with paths defined as instances of classes that
+        subclass 'bytes'.
+        """
+
+        class MyBytes(bytes):
+            pass
+
+        class SimpleComponent(component.Component):
+            class Media:
+                css = {
+                    "all": [
+                        MyBytes(b"path/to/style.css"),
+                        b"path/to/style2.css",
+                    ],
+                    "print": [
+                        MyBytes(b"path/to/style3.css"),
+                    ],
+                    "screen": b"path/to/style4.css",
+                }
+                js = [
+                    MyBytes(b"path/to/script.js"),
+                    "path/to/script2.js",
+                ]
+
+        comp = SimpleComponent()
+        self.assertHTMLEqual(
+            comp.render_dependencies(),
+            """
+            <link href="path/to/style.css" media="all" rel="stylesheet">
+            <link href="path/to/style2.css" media="all" rel="stylesheet">
+            <link href="path/to/style3.css" media="print" rel="stylesheet">
+            <link href="path/to/style4.css" media="screen" rel="stylesheet">
+
+            <script src="path/to/script.js"></script>
+            <script src="path/to/script2.js"></script>
+            """,
+        )
+
+    def test_function(self):
+        class SimpleComponent(component.Component):
+            class Media:
+                css = [
+                    lambda: mark_safe('<link hi href="calendar/style.css" rel="stylesheet" />'),  # Literal
+                    lambda: Path("calendar/style1.css"),
+                    lambda: "calendar/style2.css",
+                    lambda: b"calendar/style3.css",
+                ]
+                js = [
+                    lambda: mark_safe('<script hi src="calendar/script.js"></script>'),  # Literal
+                    lambda: Path("calendar/script1.js"),
+                    lambda: "calendar/script2.js",
+                    lambda: b"calendar/script3.js",
+                ]
+
+        comp = SimpleComponent()
+        self.assertHTMLEqual(
+            comp.render_dependencies(),
+            """
+            <link hi href="calendar/style.css" rel="stylesheet" />
+            <link href="calendar/style1.css" media="all" rel="stylesheet">
+            <link href="calendar/style2.css" media="all" rel="stylesheet">
+            <link href="calendar/style3.css" media="all" rel="stylesheet">
+
+            <script hi src="calendar/script.js"></script>
+            <script src="calendar/script1.js"></script>
+            <script src="calendar/script2.js"></script>
+            <script src="calendar/script3.js"></script>
+            """,
+        )
+
+    @override_settings(STATIC_URL="static/")
+    def test_works_with_static(self):
```

Comment:
Added a separate test here to check that the paths defined ins `Media.css` / `Media.js` work with staticfiles.